cabal.project

```diff
@@ -54,3 +54,35 @@ package nothunks
 -- (NOTE this will change to ieee754 in newer versions of nixpkgs).
 extra-packages: ieee, filemanip
 
+if impl(ghc >= 9.6)
+  allow-newer:
+    , *:base
+    , *:ghc-prim
+    , *:mtl
+    , *:template-haskell
+    , *:transformers
+    , hedgehog:*
+    , microlens-mtl:*
+    , protolude:*
+    , th-compat:*
+
+-- Version on Hackages does not compile with `ghc-9.6`
+-- https://github.com/haskell-foundation/foundation/issues/576
+source-repository-package
```

Comment:
Honestly, maybe we should just try and purge our `foundation` dep, it's a pain :thinking:

Reply:
Bah, it comes in via `cardano-crypto`, which we need for some old crypto functions. Maybe we can vendor them...

Reply:
I am chasing the `foundation` problem via other channels.